File: web-app/src/apis/cinema.ts
import {callApi} from "./callAPI";

export const changeStatusCinema = async (
    data: {
        id: number
    }
) => {
    return await callApi<any>(`cinema-service/cinema/${data.id}`, "put")
}

export const editCinema = async (
    data: {
        id: number,
        name: string,
        address: string,
        adminId: number,
    }
) => {
    return await callApi<any>(`cinema-service/cinema/edit`, "put",data)
}

File: payment-service/src/main/java/com/example/paymentservice/entity/Cinema.java
package com.example.paymentservice.entity;

import java.time.LocalDateTime;
import java.util.List;

import jakarta.persistence.*;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "cinema")
@Data
@NoArgsConstructor
public class Cinema {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "name")
    private String name;

    @Column(name = "address")
    private String address;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "admin_id", nullable = false)
    private User admin;

    @OneToMany(mappedBy = "cinema", fetch = FetchType.LAZY)
    @JsonIgnore
    private List<Room> rooms;

    @Column(name = "createdAt")
    private LocalDateTime createdAt;

    @Column(name = "lastModifyAt")
    private LocalDateTime lastModifyAt;

    @Column(name = "createdBy")
    private String createdBy;

    @Column(name = "lastModifyBy")
    private String lastModifyBy;
}
